lib/magma/actions/model_update_actions.rb

```diff
@@ -1,26 +1,71 @@
+require_relative 'base_action'
 require_relative 'update_attribute'
+require_relative 'add_attribute'
 
 class Magma
   class ModelUpdateActions
+    class FailedActionError < StandardError; end
+
     def self.build(project_name, actions_list)
       new(project_name, actions_list)
     end
 
     def perform
-      valid? && @actions.all?(&:perform)
+      return false unless valid?
+
+      Magma.instance.db.transaction do
+        raise FailedActionError unless @actions.all?(&:perform)
+        update_model_tables
+        true
+      end
+    rescue
+      @actions.each(&:rollback)
+      false
     end
 
     def errors
-      @errors + @actions.flat_map(&:errors)
+      @errors.map(&:to_h) + @actions.flat_map(&:errors)
     end
 
     private
 
+    def update_model_tables
+      migrations = @project.migrations
+      return if migrations.all?(&:empty?)
+
+      sequel_migration = eval("
```

Comment:
Calling `eval` is a little risky, but we're not running user generated input so we should be OK. It feels like a decent trade off to get the migrations running. In the future, we might want to explore building the migrations without depending on strings.